### ISSUE TITLE
Running oro:message-queue:consume is interrupted mysql errors

### ISSUE THREAD

**Issue by dimonixx** [oroinc/platform#754]
Hi 

after a curtain time `oro:message-queue:consume` is interrupted with the exception 'Too many connections'.

The same project on another environment: the same process is throws an exception with message 'Mysql has gone away', whereby no Mysql error can be found in the error.log, no exceptions in the app/logs/... .  

OroPlatform version 2.3.6

Are there any preffered mysql settings ?

**Comment by vladimirseniuk** [comment]
Hi @dimonixx

Thank you for reporting the issue!

If you get a `Too many connections error`, this means that all available connections are in use. Look at [max_connections](https://dev.mysql.com/doc/refman/5.5/en/server-system-variables.html#sysvar_max_connections) system variable.

Also, can you provide more details to clarify the problem:
 - how many consumers are you running at once?
 - how many jobs are in the queue?

**Comment by dimonixx** [comment]
Hi @vladimirseniuk 

i've tried to increase `max_connection`to 1000, did not help.
I ran 3 consumers, the `oro_message_queue` had about 20K rows.

**Comment by dimonixx** [comment]
I have a similar error in another project. The queue consumer is interrupted , when trying to reindex a search item. It fails when selecting an item from `oro_search_item`. All i can see in the logs a general MySQL error `Server has gone away`. This happens not always, but after some time of running the queue consumer.

**Comment by vtsykun** [comment]
I have the same errors on postgres after update from 3.3.2 to 3.3.6
```
[error] Consuming interrupted by exception. "An exception occured in driver: SQLSTATE[08006] [7] FATAL:  sorry, too many clients alreadyFATAL:  sorry, too many clients already"[debug] 
[RejectMessageOnExceptionDbalExtension] Execution was interrupted and message was rejected. oro.59c244e07788b3.30335162   
[Doctrine\DBAL\Exception\ConnectionException]  An exception occured in driver: SQLSTATE[08006] [7] 
FATAL:  sorry, too many clients already    FATAL:  sorry, too many clients already
```
I thinking that it related with new `ContainerResetExtension` Because after added some processors to `persistent_processors` MQ continues works stably 
```yml
# app.yml 
oro_message_queue:
    persistent_processors:
        - 'processors1'
...
```
\cc @vsoroka

**Comment by vtsykun** [comment]
Hi @vladimirseniuk I can provide STR to reproduce this bug 
**STR**
1) Install OroCRM 2.3.6, PostgreSQL 9.5
2) Set `max_connections = 100` postgresql.conf (default value)
3) Create `processes.yml`
```yml
processes:
    definitions:
        mq_test:
            label: "Test message queue"
            enabled: true
            entity: Oro\Bundle\SalesBundle\Entity\Opportunity
            preconditions:
                '@not_empty': $id
            actions_configuration:
                - '@create_note':
                    message: 'Test'
                    target_entity: $.data
                    attribute: $.result.note
                - '@create_date':
                    attribute: $.date
                - '@transit_workflow':
                    entity: $.data
                    transition: close_won
                    workflow: opportunity_flow
                    data:
                        close_date: $.date
                - '@clone_entity': # only for this STR
                    target: $.data
                    attribute: $.clone
                    flush: true

    triggers:
        mq_test:
            -
                event: create
                queued: true
                priority: 10
```
*Note* The  main goal of `clone_entity` that the work of consumer never ends (in real we use import with 50k opportunity)

4. Activate opportunity_flow
5. Create one opportunity to trigger process
6. Run consumer in prod mode

Actual
After 30-60 sec consumer stopped with errors
```
[debug] Payload: '{"process_job_id":1770}'
[info] Message processed: oro.message_queue.consumption.ack
[debug] [DoctrineClearIdentityMapExtension] Clear identity map for manager "default"
[debug] Switch to a queue oro.default
[info] Pre receive Message
[info] [ContainerResetExtension] Reset the container
[error] Consuming interrupted by exception. "An exception occured in driver: SQLSTATE[08006] [7] FATAL:  sorry, too many clients already
FATAL:  sorry, too many clients already"
[debug] [RejectMessageOnExceptionDbalExtension] Execution was interrupted and message was rejected. oro.59c774e7226ea4.28049854

                                                                                               
  [Doctrine\DBAL\Exception\ConnectionException]                                                
  An exception occured in driver: SQLSTATE[08006] [7] FATAL:  sorry, too many clients already  
  FATAL:  sorry, too many clients already                                                      
                                                                                               

Exception trace:
 () at /var/www/dev/application/crm/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractPostgreSQLDriver.php:85
 Doctrine\DBAL\Driver\AbstractPostgreSQLDriver->convertException() at /var/www/dev/application/crm/vendor/doctrine/dbal/lib/Doctrine/DBAL/DBALException.php:145
 Doctrine\DBAL\DBALException::driverException() at /var/www/dev/application/crm/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOPgSql/Driver.php:66
 Doctrine\DBAL\Driver\PDOPgSql\Driver->connect() at /var/www/dev/application/crm/vendor/doctrine/dbal/lib/Doctrine/DBAL/Connection.php:360
 Doctrine\DBAL\Connection->connect() at /var/www/dev/application/crm/vendor/doctrine/dbal/lib/Doctrine/DBAL/Connection.php:429
 Doctrine\DBAL\Connection->getDatabasePlatformVersion() at /var/www/dev/application/crm/vendor/doctrine/dbal/lib/Doctrine/DBAL/Connection.php:389
 Doctrine\DBAL\Connection->detectDatabasePlatform() at /var/www/dev/application/crm/vendor/doctrine/dbal/lib/Doctrine/DBAL/Connection.php:328
 Doctrine\DBAL\Connection->getDatabasePlatform() at /var/www/dev/application/crm/vendor/doctrine/doctrine-bundle/ConnectionFactory.php:61
 Doctrine\Bundle\DoctrineBundle\ConnectionFactory->createConnection() at /var/www/dev/application/crm/app/cache/prod/appProdProjectContainer.php:2
 appProdProjectContainer->getDoctrine_Dbal_DefaultConnectionService() at /var/www/dev/application/crm/app/bootstrap.php.cache:2226
 Symfony\Component\DependencyInjection\Container->get() at /var/www/dev/application/crm/vendor/symfony/symfony/src/Symfony/Bridge/Doctrine/ManagerRegistry.php:35
 Symfony\Bridge\Doctrine\ManagerRegistry->getService() at /var/www/dev/package/platform/src/Oro/Bundle/EntityBundle/ORM/Registry.php:29
 Oro\Bundle\EntityBundle\ORM\Registry->getService() at /var/www/dev/application/crm/vendor/doctrine/common/lib/Doctrine/Common/Persistence/AbstractManagerRegistry.php:128
 Doctrine\Common\Persistence\AbstractManagerRegistry->getConnection() at /var/www/dev/package/platform/src/Oro/Bundle/MessageQueueBundle/Consumption/Extension/DoctrinePingConnectionExtension.php:31
 Oro\Bundle\MessageQueueBundle\Consumption\Extension\DoctrinePingConnectionExtension->onPreReceived() at /var/www/dev/package/platform/src/Oro/Bundle/MessageQueueBundle/Consumption/Extension/ResettableExtensionWrapper.php:56
 Oro\Bundle\MessageQueueBundle\Consumption\Extension\ResettableExtensionWrapper->onPreReceived() at /var/www/dev/package/platform/src/Oro/Bundle/MessageQueueBundle/Consumption/Extension/ChainExtension.php:48
 Oro\Bundle\MessageQueueBundle\Consumption\Extension\ChainExtension->onPreReceived() at /var/www/dev/package/platform/src/Oro/Component/MessageQueue/Consumption/ChainExtension.php:45
 Oro\Component\MessageQueue\Consumption\ChainExtension->onPreReceived() at /var/www/dev/package/platform/src/Oro/Component/MessageQueue/Consumption/QueueConsumer.php:172
 Oro\Component\MessageQueue\Consumption\QueueConsumer->doConsume() at /var/www/dev/package/platform/src/Oro/Component/MessageQueue/Consumption/QueueConsumer.php:123
 Oro\Component\MessageQueue\Consumption\QueueConsumer->consume() at /var/www/dev/package/platform/src/Oro/Component/MessageQueue/Client/ConsumeMessagesCommand.php:69
 Oro\Component\MessageQueue\Client\ConsumeMessagesCommand->execute() at /var/www/dev/application/crm/vendor/symfony/symfony/src/Symfony/Component/Console/Command/Command.php:259
 Symfony\Component\Console\Command\Command->run() at /var/www/dev/application/crm/vendor/symfony/symfony/src/Symfony/Component/Console/Application.php:866
 Symfony\Component\Console\Application->doRunCommand() at /var/www/dev/application/crm/vendor/symfony/symfony/src/Symfony/Component/Console/Application.php:193
 Symfony\Component\Console\Application->doRun() at /var/www/dev/application/crm/vendor/symfony/symfony/src/Symfony/Bundle/FrameworkBundle/Console/Application.php:92
 Symfony\Bundle\FrameworkBundle\Console\Application->doRun() at /var/www/dev/application/crm/vendor/symfony/symfony/src/Symfony/Component/Console/Application.php:124
 Symfony\Component\Console\Application->run() at /var/www/dev/application/crm/app/console:22

                                                               
  [Doctrine\DBAL\Driver\PDOException]                          
  SQLSTATE[08006] [7] FATAL:  sorry, too many clients already  
  FATAL:  sorry, too many clients already                      
```

**Comment by vsoroka** [comment]
The issue fixed in versions 2.0.24, 2.3.9 and master
